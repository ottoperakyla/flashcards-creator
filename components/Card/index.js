import React, {useState} from 'react'
import styled from 'styled-components'

const FlipCard = styled.div`
  background-color: transparent;
  display: flex;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  width: 300px;
  height: 100px;
  border: 1px solid #f1f1f1;
  perspective: 1000px;
`

const FlipCardInner = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  text-align: center;
  transition: transform 0.8s;
  transform-style: preserve-3d;
`

const FlipCardSide = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  backface-visibility: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
`

const FlipCardFront = styled(FlipCardSide)`
  background-color: #bbb;
  color: black;
`

const FlipCardBack = styled(FlipCardSide)`
  background-color: dodgerblue;
  color: white;
  transform: rotateY(180deg);
`

export default ({term, definition, setFlipped, flipped}) => {
  return (
    <FlipCard onClick={() => setFlipped(term)}>
      <FlipCardInner style={{transform: flipped === term && 'rotateY(180deg)'}}>
        <FlipCardFront>
          <p>{term}</p>
        </FlipCardFront>
        <FlipCardBack>
          <p>{definition}</p>
        </FlipCardBack>
      </FlipCardInner>
    </FlipCard>
  )
}
