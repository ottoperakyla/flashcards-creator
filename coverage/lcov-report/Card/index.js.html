<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Card/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Card</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/10</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, {useState} from 'react'
import styled from 'styled-components'
import Button from '../Button'
&nbsp;
const FlipCard = styled.div`
  background-color: transparent;
  display: flex;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  width: 300px;
  height: 100px;
  perspective: 1000px;
  transform: scale(1);
  transition: 250ms ease-in-out;
  margin: 0 0.25rem;
&nbsp;
  &amp;:hover {
    transform: scale(1.1);
  }
`
&nbsp;
const FlipCardInner = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  text-align: center;
  transition: transform 0.8s;
  transform-style: preserve-3d;
`
&nbsp;
const FlipCardSide = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  backface-visibility: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
`
&nbsp;
const FlipCardFront = styled(FlipCardSide)`
  background-color: #bbb;
  color: black;
`
&nbsp;
const FlipCardBack = styled(FlipCardSide)`
  background-color: dodgerblue;
  color: white;
  transform: rotateY(180deg);
`
&nbsp;
export default ({term, definition, setFlipped, isFlipped, flipped, idx, editCard, deleteCard}) =&gt; {
  return (
    &lt;div&gt;
      &lt;FlipCard onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >!isFlipped </span>
        ? setFlipped(flipped.concat(term)) 
        : setFlipped(flipped.filter(<span class="fstat-no" title="function not covered" >t </span>=&gt; <span class="cstat-no" title="statement not covered" >t !== term)</span>)}&gt;
        &lt;FlipCardInner style={{transform: isFlipped &amp;&amp; 'rotateY(180deg)'}}&gt;
          &lt;FlipCardFront&gt;
            &lt;p&gt;{term}&lt;/p&gt;
          &lt;/FlipCardFront&gt;
          &lt;FlipCardBack&gt;
            &lt;p&gt;{definition}&lt;/p&gt;
          &lt;/FlipCardBack&gt;
        &lt;/FlipCardInner&gt;
      &lt;/FlipCard&gt;
      &lt;div&gt;
        &lt;Button onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >editCard(idx)}</span>&gt;Edit&lt;/Button&gt;
        &lt;Button onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >deleteCard(idx)}</span>&gt;Delete&lt;/Button&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  )
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Dec 22 2018 16:58:29 GMT+0200 (EET)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
