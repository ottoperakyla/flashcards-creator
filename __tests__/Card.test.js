import React from 'react'
import  { create } from 'react-test-renderer'
import Card from '../components/Card'

test('snapshot', () => {
  const C = create(<Card />)
  expect(C.toJSON()).toMatchSnapshot()
})