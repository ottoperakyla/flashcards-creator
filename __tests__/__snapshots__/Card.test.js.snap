// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="sc-bdVaJa eLVlXy"
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH hhcFbh"
    style={
      Object {
        "transform": "rotateY(180deg)",
      }
    }
  >
    <div
      className="sc-htpNat sc-bxivhb cFOsfN"
    >
      <p />
    </div>
    <div
      className="sc-htpNat sc-ifAKCX lnaJNh"
    >
      <p />
    </div>
  </div>
</div>
`;
